qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -265,6 +275,23 @@ private void validateDataModel(Node parent, Template template, QuteValidationSet
 				case INCLUDE:
 					validateIncludeSection((IncludeSection) section, diagnostics);
 					break;
+				case CASE:
+				case IS:
+					Section parentSection = section.getParentSection();
+					if (!isSwitchWhenSection(parentSection)) {
+						Range range = QutePositionUtility.selectStartTagName(section);
+						String expectedSection = section.getSectionKind() == SectionKind.CASE ? "switch" : "when";

```

Comment:
case can be hosted in a switch/when section. So I will change the error message to say that.